src/components/SideBar/components/useModalControl.ts

```diff
@@ -0,0 +1,41 @@
+import useDeleteGoal from '@hooks/api/goalsAPI/useDeleteGoal';
+import { useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+export default function useModalControl() {
+  const [showTodoModal, setShowTodoModal] = useState(false);
+  const [isDeletePopupVisible, setIsDeletePopupVisible] = useState(false);
+  const [goalId, setGoalId] = useState<number>(0);
+  const { mutate: deleteGoalMutate } = useDeleteGoal();
+  const onShowTodoModal = () => setShowTodoModal(true);
+  const onShowDeletePopup = (id: number) => {
+    setIsDeletePopupVisible(true);
+    setGoalId(id);
+  };
+  const navigate = useNavigate();
+  const location = useLocation();
+  const handleDelete = () => {
+    deleteGoalMutate(goalId, {
+      onSuccess: () => {
+        setIsDeletePopupVisible(false);
+        if (
+          location.pathname === `/goal-detail/${goalId}` ||
```

Comment:
아 이부분도 생각해보니까 
```suggestion
          location.pathname === `/${routes.goalDetail}/${goalId}`||
```
이런식으로 하는게 좋겠네요

Reply:
아하 넵! 반영할께요!